packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -122,6 +128,9 @@ function getRelayFileWriter(baseDir: string) {
 function getSchema(schemaPath: string): GraphQLSchema {
   try {
     let source = fs.readFileSync(schemaPath, 'utf8');
+    if (path.extname(schemaPath) === '.json') {
+      source = printSchema(buildClientSchema(JSON.parse(source).data));
+    }
```

Comment:
I don't think the compiler should directly rely on having network access, especially since many private company graphql schema endpoints require authentication which we shouldn't be building here. 

However, I think it would be a great addition to have a simple script which allows you to provide a GraphQL endpoint and produce a schema.graphql file which you could later provide to the relay compiler